cmd/icingadb/main.go

```diff
@@ -67,8 +72,38 @@ func run() int {
 		}
 	}
 
-	if err := icingadb.CheckSchema(context.Background(), db); err != nil {
-		logger.Fatalf("%+v", err)
+	{
+		err := retry.WithBackoff(
+			context.Background(),
+			func(ctx context.Context) (err error) {
+				return database.AutoUpgradeSchema(
+					context.Background(), db, cmd.Config.Database.Database, "icingadb_schema", "version", "timestamp",
+					map[string]database.SchemaData{
+						database.MySQL: {
+							Schema: []string{mysql.Schema},
+							Upgrades: []database.SchemaUpgrade{
+								{Version: "3", DDL: []string{mysql.Upgrade100}},
+								{Version: "4", DDL: []string{mysql.Upgrade111}},
+								{Version: "5", DDL: []string{mysql.Upgrade120}},
+							},
+						},
+						database.PostgreSQL: {
+							Schema: []string{pgsql.Schema},
+							Upgrades: []database.SchemaUpgrade{
+								{Version: "2", DDL: []string{pgsql.Upgrade111}},
+								{Version: "3", DDL: []string{pgsql.Upgrade120}},
```

Comment:
* @lippserd This is a raw\* usage example of https://github.com/Icinga/icinga-go-library/pull/15

Do you consider the design fine?

\* One has to maintain such schema file lists. A non-raw usage requires the specific project to have a function which auto-calculates schema upgrade lists from the embedded tree of schema/.